acceptancetests/clean_prodstack.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+
+import re
+import os
+import sys
+import logging
+import subprocess
+
+def get_current_controller_and_models():
+    lines = subprocess.check_output(
+        ['juju', 'models'], stderr=subprocess.STDOUT).split('\n')
+    ctrl_name = [
+        item.split(':')[-1].strip() for item in lines if 'Controller:' in item][0]
+    for i in range(0, len(lines)):
+        if ('Model' in lines[i] and
+            'Cloud/Region' in lines[i] and
+            i+1<len(lines)):
+            model_lines = filter(None, lines[i+1:])
+            break
+    if not model_lines:
+        logging.info('No model has been found, exit.')
+        sys.exit(1)
+    model_list = [elem.split(' ')[0].strip('*') for elem in model_lines]
+    logging.info(
+        'Models under the Controller {} are:\n{}'.format(
+        ctrl_name,
+        '\n'.join(model_list)))
+    return (ctrl_name, model_list)
+
+
+def get_reserved_machine_ipaddress():
+    role = 'admin'
+    ip_pool = []
+    ctrl_name, model_list = get_current_controller_and_models()
+    for item in model_list:
+        model = '{}:{}/{}'.format(ctrl_name, role, item)
+        lines = subprocess.check_output(
+            ['juju', 'show-machine', '--model', model],
+            stderr=subprocess.STDOUT)
+        # match ipv4 address, no validation as it comes from Juju
+        ip_list = re.findall(r'\b\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\b', lines)
+        if ip_list:
+            [ip_pool.append(elem) for elem in ip_list if elem not in ip_pool]
+    if ip_pool:
+        logging.info(
+            'DO NOT touch servers associated to these ip address:\n{}'.format(
+            '\n'.join(ip_pool)))
+    else:
+        logging.info(
+            'No ip address has been found, free to delete all nova servers.')
+    return ip_pool
+
+
+def get_target_server_list():
+    ip_pool = get_reserved_machine_ipaddress()
+    egrep_out = []
+    reserved_nova_server_dict = {}
+    full_nova_server_dict = {}
+    target_nova_server_dict = {}
+    temp_file = '/tmp/burton-test-nova-list.txt'
+    with open(temp_file, 'w') as f:
+        subprocess.check_call(['nova', 'list'], stdout=f)
+    if not os.path.isfile(temp_file):
+        logging.info('Failed to get nova server list, exit.')
+        sys.exit(1)
+    with open(temp_file, 'r') as stream:
+        # using readlines() as the outout from nova list is tiny
+        for line in stream.readlines():
+            if 'juju-' in line:
+                full_nova_server_dict[
```

Comment:
This needs a comment as to what it's doing, it's not obvious what output it's trying to coerce into what format.

I *think* using "nova show --minimal" should simplify this a little (just shows uuid and name)